core/src/main/java/org/apache/iceberg/catalog/TableCommit.java

```diff
@@ -26,6 +26,12 @@
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.immutables.value.Value;
 
+/**
+ * This represents a commit to be applied for a single table with {@link UpdateRequirement}s to be
+ * validated and {@link MetadataUpdate}s that have been applied. The {@link UpdateRequirement}s and
+ * {@link MetadataUpdate}s can be derived from table's base and updated {@link TableMetadata} when
+ * using {@link TableCommit#create(TableIdentifier, TableMetadata, TableMetadata)}.
+ */
```

Comment:
Can you also add Javadoc for the `create` method? It has requirements for the `TableMetadata` objects that should be noted.

Reply:
I've added the Javadoc to make this clearer when using the `create` method